DECISIONS.md

```diff
@@ -43,3 +43,27 @@
   all-fields search, add a cross-list field to the advanced search interface,
   and include cross-list classification in shortcut routes and the advanced
   interface's classification filter (with option to exclude).
+
+- 2019-04-19. Werkzeug has its own versions of url parsing/quoting functions in
+  ``urllib.parse`` (e.g. to handle MultiDicts). In previous versions we were
+  using these to build/rebuild query URLs in search results, in
+  `search.routes.ui.context_processors`. These functions are slow, however,
+  running more than 3x more slowly than their built-in predecessors. If we
+  are using ``urlencode`` on a ``MultiDict``, we can simply do
+  ``urlencode(data.items(multi=True))``. Going forward, we will use only the
+  built-in functions in ``urllib.parse`` unless there is a very strong reason
+  to do otherwise.
+
+- 2019-04-22. While convenient, dataclasses have significant performance
+  overhead. Specifically, casting dataclasses from/to native Python structs
+  runs around two orders of magnitude more costly than just initializing new
+  dicts. This is fine if the number of instances is low. But in this case, we
+  were using dataclasses to represent individual search results, and casting
+  to/from dataclass representations without a whole lot of functional benefit.
+  For this reason, we are using dataclasses only for "low-volume" structs like
+  queries. For search results, we are now using
+  [TypedDicts](https://www.python.org/dev/peps/pep-0589/). This gives us type
+  checking machinery, is consistent with the [data domain](https://arxiv.github.io/arxiv-arxitecture/crosscutting/services.html#data-domain)
+  concept in the arXiv service architecture, and is considerably more
+  performant. In the future, we may want to consider moving entirely to
+  TypedDict for consistency.
```

Comment:
Thank you for updating the `DECISIONS.md`. This made it much easier to follow the PR.